.azure-pipelines/pipelines.yml

```diff
@@ -415,9 +415,44 @@ stages:
               artifactName: windows.release
             condition: always()
 
-      - job: docker
+      - job: clang_cl
         dependsOn: ["release"]
         timeoutInMinutes: 120
+        pool:
+          vmImage: "windows-latest"
+        steps:
+          - task: Cache@2
+            inputs:
+              key: '"windows.clang-cl" | ./WORKSPACE | **/*.bzl'
```

Comment:
@sunjayBhatia do we want to use the same key here as the other build, e.g. share a cache, or does the clang-cl cache need to be different? I'm guessing if they are the repository objects, sharing them is desired?

Reply:
Thats a good question, I'm not sure if anything build specific goes in there but as you say, makes sense if it didn't

@lizan Do you have any reasoning behind why the different Linux jobs have separate cache entries? https://github.com/envoyproxy/envoy/blob/7653adfd88379586cc55518b0548ba88fbea6dae/.azure-pipelines/bazel.yml#L24

Reply:
I'll leave this as-is, a distinct cache for clang-cl, pending lizan's feedback.

Reply:
I split this because different build targets are pulling slightly different set of repositories, because repositories are lazily pulled. For example the compile_time_options will pull boringssl_fips instead of normal boringssl, additional wasm runtimes such as wasmtime/wavm.

So if clang-cl and msvc-cl shares exact same set of repositories, they don't have to be a different key, if not we can keep them different. Up to you two.